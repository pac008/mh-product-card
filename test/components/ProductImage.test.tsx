import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard} from '../../src/components';
import { product2 } from '../data/products';


describe('Test in productImage', () => {
  test('Should display the component correctly with the custom image', () => {
    const wrapper = renderer.create(
        <ProductImage img='custom-product-image' className='custom-class' />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should display component with the product name', () => { 
      
      const wrapper = renderer.create(
        <ProductCard product={product2}>
            {
                () => (
                    <ProductImage  />
                )
            }
        </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
   })
});
