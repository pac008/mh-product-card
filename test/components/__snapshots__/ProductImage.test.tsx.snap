// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in productImage Should display the component correctly with the custom image 1`] = `
<img
  alt="Coffe Mug"
  className="productImg custom-class "
  src="custom-product-image"
/>
`;

exports[`Test in productImage should display component with the product name 1`] = `
<div
  className="productCard undefined "
>
  <img
    alt="Coffe Mug"
    className="productImg undefined "
    src="./coffee-mug.png"
  />
</div>
`;

exports[`Test in productTitle Should display the component correctly with the custom image 1`] = `
<img
  alt="Coffe Mug"
  className="productImg custom-class "
  src="custom-product-image"
/>
`;

exports[`Test in productTitle should display component with the product name 1`] = `
<div
  className="productCard undefined "
>
  <img
    alt="Coffe Mug"
    className="productImg undefined "
    src="./coffee-mug.png"
  />
</div>
`;
